pkg/solana/txm/txm.go

```diff
@@ -781,6 +782,28 @@ func (txm *Txm) Enqueue(ctx context.Context, accountID string, tx *solanaGo.Tran
 		return fmt.Errorf("failed to store new transaction with ID %s: %w", id, err)
 	}
 
+	// If a dependency transaction ID is provided, handle waiting and enqueuing asynchronously.
+	if cfg.DependencyTxID != "" {
+		go func(msg pendingTx, depID string) {
+			err := txm.waitForTxStatus(ctx, depID, types.Unconfirmed)
```

Comment:
Since we are not on critical path, maybe we could wait for Finalized state?

Reply:
Good point, updated.